resolvers/event.js

```diff
@@ -17,41 +36,230 @@ const eventResolvers = {
+        });
+        
+        // Adding in records of a singular association with the event and the hospital
+        for (const hospitalId of args.hospitals) {
+          await db.eventHospitals.create({
+            eventId: args.id,
+            hospitalId: hospitalId['id']
+          })
+        }
+
+        return false;
+      }
+
+      if (args.hospitals) {
+        // Checking if hospitals are valid
+        const hospitalHasError = await addHospitals();
+        if (hospitalHasError) {
+          throw new Error("Invalid hospital ID");
+        }
+
       }
 
       await db.event.update(
         {
           name: args.name,
           eventDate: args.eventDate,
           createdBy: args.createdBy,
-          isActive: args.isActive
+          isActive: args.isActive,
         },
         {
           where: { id: args.id }
         }
       );
-      return db.event.findByPk(args.id);
+      return db.event.findByPk(args.id, { 
+        include: [{
+          model: db.ambulance,
+          attributes: ['id', 'vehicleNumber', 'createdAt', 'updatedAt'],
+        },
+        {
+          model: db.hospital,
+          attributes: ['id', 'name', 'createdAt', 'updatedAt']
+        }]
+      });
+    },
+    addAmbulanceToEvent: async (parent, args) => {
+      // Checking if event exists
+      const event = await db.event.findByPk(args.eventId);
+      if (!event) {
+        throw new Error("Invalid event ID");
+      }
+
+      // Checking if ambulance exists
+      const ambulance = await db.ambulance.findByPk(args.ambulanceId['id']);
+      if (!ambulance) {
+        throw new Error("Invalid ambulance ID");
+      }
+      
+      // Checking if association between ambulance and event already exists
+      const alreadyExists = await db.eventAmbulances.count({
+        where: {
+          eventId: args.eventId,
+          ambulanceId: args.ambulanceId['id']
+        }
+      })
+
+      if (alreadyExists > 0) {
+        throw new Error("This ambulance is already assigned to this event")
+      }
+
+      // Creating association in eventAmbulances junction table
+      await db.eventAmbulances.create({
+        eventId: args.eventId,
+        ambulanceId: args.ambulanceId['id']
+      })
+
+      // Returning new event
+      return db.event.findByPk(args.eventId, { 
+        include: [{
+          model: db.ambulance,
+          attributes: ['id', 'vehicleNumber', 'createdAt', 'updatedAt'],
+        },
+        {
+          model: db.hospital,
+          attributes: ['id', 'name', 'createdAt', 'updatedAt']
+        }]
+      });
     },
-    deleteEvent: (parent, args) => {
+    addHospitalToEvent: async (parent, args) => {
+      // Checking if event exists
+      const event = await db.event.findByPk(args.eventId);
+      if (!event) {
+        throw new Error("Invalid event ID");
+      }
+      
+      // Checking if hospital exists
+      const hospital = await db.hospital.findByPk(args.hospitalId['id']);
+      if (!event) {
```

Comment:
```suggestion
      if (!hospital) {
```

Reply:
I believe due to this small error I can add non-existent hospitals
![image](https://user-images.githubusercontent.com/38408276/84604808-b385a980-ae66-11ea-8228-fc030be1bd82.png)
![image](https://user-images.githubusercontent.com/38408276/84604812-b97b8a80-ae66-11ea-8a70-a788ae75a7d8.png)
Other than that looks good! Great job guys :)